app/src/main/java/com/example/bookstore/fragment/action/ActionOnJobOfDay.kt

```diff
@@ -0,0 +1,147 @@
+
+        ivDot = holder.itemView.findViewById(R.id.iv_dot)
+
+        txtJobName = holder.itemView.findViewById(R.id.txt_job_name)
+
+        setDataInJob(position, data)
+        setIconCompleted(data[position])
+        setColorInStatus(holder, data[position])
+    }
+
+    @SuppressLint("ResourceAsColor")
+    private fun setColorInStatus(holder: RecyclerAdapter<Job>.ViewHolder, job: Job) {
+
+        when (job.status) {
+            JobStatus.MISSED -> {
+                cvJob.setCardBackgroundColor(context.colorList(R.color.c_f7f8fc))
+                txtStatus.setTextColor(context.colorList(R.color.red))
+            }
+
+            JobStatus.COMPLETED -> {
+                cvJob.setCardBackgroundColor(context.colorList(R.color.c_7470ef))
+                setColorForTextJob(holder, R.color.white)
+
+                txtExercise.isVisible = false
+                ivDot.isVisible = false
+
+                imgTickCompleted.apply {
+                    setImageResource(R.drawable.completed_icon)
+                    setOnClickListener(null)
+                    isClickable = false
+                }
+            }
+
+            else -> setColorForTextJob(holder, R.color.silver)
+        }
+    }
+
+    @SuppressLint("SetTextI18n")
+    private fun setDataInJob(
+        position: Int,
+        jobs: List<Job>
+    ) {
+        txtJobName.text = jobs[position].name
+
+        val status = jobs[position].status
+        if (status != JobStatus.ASSIGNED) {
+            txtStatus.text =
+                status.name.substring(0, 1) + status.name.substring(1, status.name.length)
+                    .lowercase(Locale.ROOT)
+        } else {
+            ivDot.isVisible = false
+        }
+
+        when (jobs.size) {
+            0 -> txtExercise.text = ""
+            1 -> txtExercise.text = "${jobs[position].exercises} exercise"
+            else -> txtExercise.text = "${jobs[position].exercises} exercises"
+        }
+    }
+
+    @SuppressLint("ResourceAsColor")
+    private fun onClickCompleteJob(
+        imgView: ImageView,
+        job: Job
+    ): View.OnClickListener {
+        return View.OnClickListener {
+            isSelected = imgView.tag.toString().toBooleanStrict()
+            if (!isSelected) {
+                imgView.setImageResource(R.drawable.completed_icon_purple)
+            } else {
+                imgView.setImageResource(R.drawable.completed_icon)
+            }
+            imgView.tag = !isSelected
+
+            job.isSelected = !isSelected
+            calendarViewModel.updateIsSelected(job)
+        }
+    }
+
+    private fun setIconCompleted(job: Job) {
+        if (job.isSelected) {
+            imgTickCompleted.setImageResource(R.drawable.completed_icon_purple)
+        } else {
+            imgTickCompleted.setImageResource(R.drawable.completed_icon)
+        }
+        imgTickCompleted.tag = job.isSelected
+    }
+
+    private fun setColorForTextJob(holder: RecyclerAdapter<Job>.ViewHolder, color: Int) {
```

Comment:
Không chắc cái nào OK nhưng khi khai báo hàm mà tham số phải được refer từ Resource thì phải thêm annotation vào


```suggestion
    private fun setColorForTextJob(holder: RecyclerAdapter<Job>.ViewHolder, @ColorRes color: Int) {
```

Hoặc

```suggestion
    private fun setColorForTextJob(holder: RecyclerAdapter<Job>.ViewHolder, @ColorInt color: Int) {
```